### ISSUE TITLE
vSphere at times return null data

### ISSUE THREAD

**Issue by srazin** [yavijava/yavijava#221]
I do not believe it is a yavijava problem, but that's the closest forum I know to discuss and query for opinion that is directly deals with yavijava.
It seems that at time when we send requests through yavijava to vCenter (6.x as well as 5.x) very rearly there could be an occurrence that vCenter would return null (nothing). I can stop my process, etc, but not until I reboot the vCenter it does not come back to respond with the right information. On the other hand it seems that if I change the endpoint IP, it would start responding correctly. Maybe has something to do with the session on the vCenter side...?

Anyone seen that? Thoughts?

**Comment by MAPerk** [comment]
when I was doing some development with vijava a couple years ago, I had a situation where I was sending many requests for processing, cloning mostly, at the server and I noticed that after reaching a certain threshold I would start to get failures on the requests.  Too many sessions is what it seemed to boil down to.  I had to change the method I was using to authenticate and perform tasks. I needed to re-authenticate to the server if I went more than 20 minutes without doing something with the vCenter.

**Comment by michaelrice** [comment]
Have you checked the vCenter logs? Normally if you're hitting the limit you will see `SOAP session count limit reached` in your vCenter logs. If you are seeing that in the logs you can up the default limit. You can also increase the session timeout as well. I suggest reusing the same session for as long as possible and upping the connection limit based on your needs. Take a look at this KB article and see if it helps: https://kb.vmware.com/selfservice/microsites/search.do?language=en_US&cmd=displayKC&externalId=2004663

**Comment by srazin** [comment]
No, I have not. But have you observed something like that prior yourself?

**Comment by michaelrice** [comment]
Any time I have issues with vCenter the first place I go is the vCenter logs. Ive run into all kinds of issues in the past but I dont recall if this issue you are describing was one of them. On our production vCenters we have done *all kinds* of performance tuning though. A lot of the default settings dont work in an environment as large as ours, and I know for a fact we have extended the session timeouts and increased the session count limit.

**Comment by srazin** [comment]
Thank you for the info! I will look into it.

BTW are you guys (considering the scale at which you operate) willing to
share the setting that you found to work best? (Assuming it is consolidated
into a doc etc.)
I am sure community can learn a lot from that...
However I understand if otherwise.

**Comment by michaelrice** [comment]
This is a vCenter configuration issue rather than a yavijava bug. The most common cause is hitting the SOAP session count limit — check your vCenter logs for `SOAP session count limit reached`. See the VMware KB article linked above for tuning session count and timeout settings. Closing as not a library issue.
